app/presenters/users/devices/devices_index_presenter.rb

```diff
@@ -0,0 +1,42 @@
+module Users::Devices
+  class DevicesIndexPresenter < ApplicationPresenter
+    attr_reader :user
+    attr_reader :devices
+
+    def initialize(user, params)
+      @user = user
+      @params = params
+      @devices = index_devices
+    end
+
+    def index_devices
+      user_devices.geocode_last_checkins
+      device_ids = user_devices.last_checkins.map { |checkin| checkin["device_id"] }
+      devices = user_devices.index_by(&:id).values_at(*device_ids)
```

Comment:
We want all devices to be shown but we want devices to be in ordered by last checkin. So the device that you used most recently (i.e. with the most recent checkin) is shown first.